astropy/coordinates/sky_coordinate.py

```diff
@@ -900,30 +715,30 @@ def __getattr__(self, attr):
 
     def __setattr__(self, attr, val):
         # This is to make anything available through __getattr__ immutable
-        if "_sky_coord_frame" in self.__dict__:
-            if self._is_name(attr):
-                raise AttributeError(f"'{attr}' is immutable")
+        if attr != "info" and not attr.startswith("_"):
```

Comment:
The old `__setattr__` caused me a lot of trouble, with its setting by default on the frame rather than on the coordinate instance itself. This ensures that `info` does not get propagated down, and also does a bit more sanity checks before proceeding.